website/docs/pipelines/promoting-applications.mdx

```diff
@@ -140,118 +142,170 @@ is upgraded in one of the pipeline's environments, a PR is created that updates
 
-### Define promotion in pipeline resource
+>An RBAC Role or ClusterRole contains rules that represent a set of permissions.
+> Permissions are purely additive (there are no "deny" rules).
 
-The field `.spec.promotion.strategy.pull-request` defines details about the Git repository used for promoting the given app.
-Set the `secretRef.name` field to the name of the Secret created in the previous step and the `url` and `branch` fields to the
-Git repository's HTTPS URL and optionally a specific branch (if the branch is not set, it defaults to `main`).
-If using the `generic-hmac` Provider from leaf clusters, also set the `.spec.promotion.strategy.secretRef.name` to the name of the Secret created previously.
+You should ensure that attempts to break this contract are blocked and detected. You could achieve it
+by leveraging [policy capabilities](../../policy/intro/). Policy Agent as:
+- [Admission Controller](../../policy/intro#admission-controller) protects from any attempt to create RBAC resources ending granting access to the secret.
+- [Audit](../../policy/intro#audit) helps you identify already existing resources that are out of compliance.
+For example, roles created before policy agent was introduced as admission controller.
 
-More info in the [spec](../spec/v1alpha1/pipeline/#pipeline)
+Once enabled [policy and policy library](../../policy/weave-policy-profile/) you have a set of available policies to
+leverage. `RBACProhibitWildcards` and `RBACProhibitVerbsOnResources` are the building blocks that you could use
+to implement the expected behaviour:
 
-### Security Recommendations
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- policies/RBACProhibitWildcards/no-wildcard-on-resources-policy.yaml
+- policies/RBACProhibitWildcards/no-wildcard-on-verbs-policy.yaml
+- policies/RBACProhibitVerbsOnResources/no-list-secrets-policy.yaml
+- policies/RBACProhibitVerbsOnResources/no-watch-secrets-policy.yaml
+- policies/RBACProhibitVerbsOnResources/no-wildcard-secrets-policy.yaml
+```
 
-:::tip
+An example of policy to reject using wildcards (in this case using wildcards on verbs) is shown below:
 
-Adopt as much of the recommendations in this section to reduce the risks associated with the secrets involved in pull requests.
+```yaml
+apiVersion: pac.weave.works/v2beta2
+kind: Policy
+metadata:
+  name: weave.templates.rbac-prohibit-wildcards-verbs
+spec:
+  id: weave.templates.rbac-prohibit-wildcards-verbs
+  name: Rbac Prohibit Wildcards on Verbs
+  ...
+  category: weave.categories.access-control
+  severity: high
+  parameters:
+    - name: attributes
+      type: string
+      required: true
+      value: "verbs"
+    - name: exclude_role_name
+      type: string
+      required: true
+      value: "wego-admin"
+    - name: exclude_label_key
+      type: string
+      required: false
+      value: "app.kubernetes.io/part-of"
+    - name: exclude_label_value
```

Comment:
extend 

1) be careful with rolebindings out of the namespace - for example, allowing, extending 
2) dont give kubectl access with rbac write access because as a user i could escape this control

Reply:
>dont give kubectl access with rbac write access because as a user i could escape this control

added